TLM/TLM/Custom/AI/CustomTramBaseAI.cs

```diff
@@ -713,62 +1082,98 @@ public class CustomTramBaseAI : TramBaseAI { // TODO inherit from VehicleAI (in
+
+            // base.SimulationStep(vehicleID, ref vehicleData, ref frameData, leaderID, ref leaderData, lodPhysics);
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void UpdatePathTargetPositions(TramBaseAI tramBaseAI, ushort vehicleID, ref Vehicle vehicleData, Vector3 refPos1, Vector3 refPos2, ushort leaderID, ref Vehicle leaderData, ref int index, int max1, int max2, float minSqrDistanceA, float minSqrDistanceB) {
-            Log.Error($"CustomTramBaseAI.InvokeUpdatePathTargetPositions called! tramBaseAI={tramBaseAI}");
+        [UsedImplicitly]
+        public static void UpdatePathTargetPositions(TramBaseAI tramBaseAi,
+                                                     ushort vehicleId,
+                                                     ref Vehicle vehicleData,
+                                                     Vector3 refPos1,
+                                                     Vector3 refPos2,
+                                                     ushort leaderId,
+                                                     ref Vehicle leaderData,
+                                                     ref int index,
+                                                     int max1,
+                                                     int max2,
+                                                     float minSqrDistanceA,
+                                                     float minSqrDistanceB) {
+            Log._DebugOnlyError($"CustomTramBaseAI.InvokeUpdatePathTargetPositions called! " +
+                                $"tramBaseAI={tramBaseAi}");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static float GetMaxSpeed(ushort leaderID, ref Vehicle leaderData) {
-            Log.Error("CustomTrainAI.GetMaxSpeed called");
+        [UsedImplicitly]
+        private static float GetMaxSpeed(ushort leaderId, ref Vehicle leaderData) {
+            Log._DebugOnlyError("CustomTrainAI.GetMaxSpeed called");
             return 0f;
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
+        [UsedImplicitly]
         private static float CalculateMaxSpeed(float targetDist, float targetSpeed, float maxBraking) {
-            Log.Error("CustomTrainAI.CalculateMaxSpeed called");
+            Log._DebugOnlyError("CustomTrainAI.CalculateMaxSpeed called");
             return 0f;
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void InitializePath(ushort vehicleID, ref Vehicle vehicleData) {
-            Log.Error("CustomTrainAI.InitializePath called");
+        [UsedImplicitly]
+        private static void InitializePath(ushort vehicleId, ref Vehicle vehicleData) {
+            Log._DebugOnlyError("CustomTrainAI.InitializePath called");
```

Comment:
need to call Error, not _DebugOnlyError

Reply:
in Release the function is not eliminated, the Log call is eliminated